pkg/api/testapi/testapi.go

```diff
@@ -24,63 +24,144 @@ import (
 
 	"k8s.io/kubernetes/pkg/api/latest"
 	"k8s.io/kubernetes/pkg/api/meta"
+	apiutil "k8s.io/kubernetes/pkg/api/util"
+	explatest "k8s.io/kubernetes/pkg/expapi/latest"
 	"k8s.io/kubernetes/pkg/runtime"
 )
 
-// Version returns the API version to test against, as set by the KUBE_API_VERSION env var.
-func Version() string {
-	version := os.Getenv("KUBE_API_VERSION")
-	if version == "" {
-		version = latest.Version
+var All = make(map[string]TestGroup)
+var Default TestGroup
+var Experimental TestGroup
+
+type TestGroup struct {
+	Group string
+	// in the format of "group/version"
+	GroupVersionUnderTest string
+}
+
+func init() {
+	kubeTestAPI := os.Getenv("KUBE_TEST_API")
+	if kubeTestAPI != "" {
+		testGroupVersions := strings.Split(kubeTestAPI, ",")
+		for _, groupVersion := range testGroupVersions {
+			// TODO: caesarxuchao: the apiutil package is hacky, it will be replaced
+			// by a following PR.
+			All[apiutil.GetGroup(groupVersion)] =
+				TestGroup{apiutil.GetGroup(groupVersion), groupVersion}
+		}
+	}
+
+	// TODO: caesarxuchao: we need a central place to store all available API
+	// groups and their metadata.
+	if _, ok := All[""]; !ok {
+		// TODO: will be latest.GroupVersion after we have multiple group
+		// support
+		All[""] = TestGroup{"", latest.Version}
 	}
-	return version
+	if _, ok := All["experimental"]; !ok {
+		All["experimental"] = TestGroup{"experimental", explatest.Version}
+	}
+
+	Default = All[""]
+	Experimental = All["experimental"]
+}
+
+// Version returns the API version to test against, as set by the KUBE_TEST_API env var.
+func (g TestGroup) Version() string {
+	return apiutil.GetVersion(g.GroupVersionUnderTest)
+}
+
+// GroupAndVersion returns the API version to test against for a group, as set
+// by the KUBE_TEST_API env var.
+// Return value is in the form of "group/version".
+func (g TestGroup) GroupAndVersion() string {
+	return g.GroupVersionUnderTest
 }
 
 // Codec returns the codec for the API version to test against, as set by the
-// KUBE_API_VERSION env var.
-func Codec() runtime.Codec {
-	interfaces, err := latest.InterfacesFor(Version())
-	if err != nil {
-		panic(err)
+// KUBE_TEST_API env var.
+func (g TestGroup) Codec() runtime.Codec {
+	if g.Group == "" {
```

Comment:
Ah, I did not notice the difference, sorry! This is fine then, maybe you can add a TODO about fixing it when we have a central `latest`.